crates/runtime/src/tools/builtin/sample/distinct.rs

```diff
@@ -129,26 +130,40 @@ impl SampleFrom for DistinctColumnsParams {
 
         let schema = provider.schema();
 
-        let columns = schema
-            .fields()
-            .iter()
-            .map(|f| f.name().clone())
-            .collect::<Vec<String>>();
+        let columns = schema.fields();
 
         let mut result: Vec<ArrayRef> = Vec::with_capacity(columns.len());
 
         for (i, column) in columns.iter().enumerate() {
-            // Only sample distinctly from columns that are specified in the `cols` field, or if `cols` is None.
-            let column_data = if self.cols.is_none()
-                || self.cols.as_ref().is_some_and(|cols| cols.contains(column))
+            // Only sample distinctly from columns that are specified in the `cols` field, if `cols` is None and distinct sampling is supported
+            let column_data = if column_supports_distinct_sampling(column)
+                && (self.cols.is_none()
+                    || self
+                        .cols
+                        .as_ref()
+                        .is_some_and(|cols| cols.contains(column.name())))
             {
-                Self::sample_distinct_from_column(Arc::clone(&df), &tbl, column, self.limit).await?
+                Self::sample_distinct_from_column(Arc::clone(&df), &tbl, column.name(), self.limit)
+                    .await?
             } else {
-                Self::sample_from_column(Arc::clone(&df), &tbl, column, self.limit).await?
+                Self::sample_from_column(Arc::clone(&df), &tbl, column.name(), self.limit).await?
```

Comment:
I know this is from the prior PR #3108 , but I don't understand why we still sample the column even if we explicitly decide to not sample the column ?

Reply:
Good catch this will be improved as incremental PR.